### PR TITLE
#2908 Make ModuleType a string to allow easy extensions

### PR DESCRIPTION
**Pull request by LukaszRozmej**
What RPC modules we should have enabled by default? Right now its ALL. I would say that:
Eth, Net, Parity, Personal, Proof, Subscribe, Trace, TxPool, Web3, Health

Built in modules:
Admin,
Clique,
Consensus,
Db,
Debug,
Erc20,
Eth,
Evm,
Mev,
NdmProvider,
NdmConsumer,
Net,
Nft,
Parity,
Personal,
Proof,
Subscribe,
Trace,
TxPool,
Web3,
Baseline,
Vault,
Deposit,
Health

Will fix test after this is discussed.

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Modules/ModuleType.cs [hunk 1]
```diff
@@ -14,47 +14,78 @@
...
+            Web3,
+            Baseline,
+            Vault,
+            Deposit,
+            Health
+        };
 
-        public RpcModuleAttribute(ModuleType moduleType)
+        public static IEnumerable<string> DefaultModules { get; } = new List<string>()
         {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.JsonRpc/Modules/ModuleType.cs:78]: this should be reviewed with Micah and Mystic

File: src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+
+namespace Nethermind.JsonRpc.Modules
+{
+    public class RpcModuleAttribute : Attribute
+    {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleAttribute.cs:23]: created but not used in the end?
I have been thinking about this for a long time
never liked the ModuleType enum so glad that there is an attempt to remove it
I remember it was causing some issues though in the ModuleLoader (when resolving rpc method we check whether the module is enabled or unknown and define the error message based on it)

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleAttribute.cs:23]: It was just moved to separate file. Modules can be registered by string now.

**Comment by MicahZoltu** [comment]
As long as a plugin can add an arbitrary unknown module type then I'm happy with this change.  Previously, to work around this I had to add a *numbered* module which was really annoying:
```
JsonRpc.EnabledModules 'Eth,Trace,100'
```

**Comment by tkstanczak** [comment]
> As long as a plugin can add an arbitrary unknown module type then I'm happy with this change. Previously, to work around this I had to add a _numbered_ module which was really annoying:
> 
> ```
> JsonRpc.EnabledModules 'Eth,Trace,100'
> ```

so creative...

**Review by SebastianDremo — APPROVED** [review]

**Pull request merged**